src/ConfigCatProvider.tsx

```diff
@@ -15,39 +17,33 @@ type ConfigCatProviderProps = {
 };
 
 type ConfigCatProviderState = {
-  client: IConfigCatClient;
+  client?: IConfigCatClient;
   lastUpdated?: Date;
 };
 
 const initializedClients = new Map<string, number>();
 
-class ConfigCatProvider extends Component<
-/* eslint-disable @typescript-eslint/indent */
-  PropsWithChildren<ConfigCatProviderProps>,
-  ConfigCatProviderState,
-  {}
-/* eslint-enable @typescript-eslint/indent */
-> {
+class ConfigCatProvider extends Component<PropsWithChildren<ConfigCatProviderProps>, ConfigCatProviderState, {}> {
 
   constructor(props: ConfigCatProviderProps) {
     super(props);
-    const client = this.initializeConfigCatClient();
-    this.state = { client };
+    this.state = {};
   }
 
   componentDidMount(): void {
-    this.state?.client?.on("clientReady", () => this.clientReady());
-    this.state?.client?.on("configChanged", newConfig => this.reactConfigChanged(newConfig));
+    this.setState({ client: this.initializeConfigCatClient(), lastUpdated: new Date() });
+    this.state.client?.on("clientReady", () => this.clientReady());
```

Comment:
Tangential, but at this point it's too late to subscribe to the `clientReady` hook. In many cases it already fires during the construction of the client object.

If we don't want to miss the event, we have to subscribe to it via the client options. (But even then we'd miss it when `initializeConfigCatClient` returns an already created client.)

If the `clientReady` event needs to reported reliably, we need to rework this part.